backend/core/src/main/java/org/sonarsource/sonarlint/core/fs/ClientFile.java

```diff
@@ -123,6 +126,11 @@ public void setClean() {
     this.clientProvidedContent = null;
   }
 
+  public boolean isSonarlintConfigurationFile() {
+    var sonarlintPath = getClientRelativePath().getParent();
+    return "json".equals(FileNameUtils.getExtension(getClientRelativePath())) && sonarlintPath != null && SONARLINT_FOLDER_NAME.equals(sonarlintPath.getFileName().toString());
```

Comment:
Yes, VS names the files by either `<folderName>.json` or `<solutionName>.json`, and Rider will also call .NET project config files with `<solutionName>.json`

Reply:
Yes, the backend does not care about the filename because of the specificities of C# projects - we store the module name in the name, whereas in other languages we'll store them as connectedMode.json (or something similar) to make it clearer in the VCS.

Later, the files are parsed, and if the parsing does not work we skip the file